ios/Trifle/Sources/TrifleCertificate.swift

```diff
@@ -6,11 +6,36 @@
 import Foundation
 import Wire
 
-public class TrifleCertificate : Codable {
+public class TrifleCertificate {
     private let proto: Certificate
     
-    public init(data: Data) throws {
-        self.proto = try ProtoDecoder().decode(Certificate.self, from: data)
+    /**
+     Constructor is private. This class can be constructed only via the deserialization
+     */
+    private init(proto: Certificate) {
+        self.proto = proto
+    }
+    
+    /**
+     Deserializes the data and returns a TrifleCertificate object.
+     
+     This is a static method.
+
+     - parameters: data - the serialized TrifleCertificate object
+     
+     - returns: TrifleCertificate object
+     */
+    public static func deserialize(data: Data) throws -> TrifleCertificate {
+        return TrifleCertificate( proto: try ProtoDecoder().decode(Certificate.self, from: data))
+    }
+    
+    /**
+     Serializes the TrifleCertificate object into Data object.
+     
+     - returns: Data object
+     */
+    public func serialize() throws -> Data {
+        return try ProtoEncoder().encode(proto)
     }
     
     internal func getCertificate() -> Certificate {
```

Comment:
In the case of `TrifleCertificate` we have a deliberate accessor to the underlying proto. However, for both `TrifleCertificateRequest` and `TrifleSignedData` the pattern differs. What would be the right level of exposure? Should we not be exposing the underlying proto? 

See usages in `CertificateTests` for the asymmetric data pattern employed.

Reply:
I think for this PR we should focus on changing the interface rather than how it is used under the hood.  If we have different patterns for how the underlying protos are used that should be a separate change.

Reply:
We need the accessor because we need to operate on the certificates for `TrifleCertificate#verify`. For both TrifleSignedData and TrifleCertificateRequest, the server operates on the data and so gets are not needed.
Ideally we would not expose any of the underlying protos, however in this case we need an internal accessor.

Does that make sense?